django_video_encoder/checks.py

```diff
@@ -0,0 +1,36 @@
+from django.conf import settings
+from django.core import checks
+
+from django_video_encoder.models import Format
+
+
+@checks.register
+def check__formats(app_configs, **kwargs):
+    if not hasattr(settings, "DJANGO_VIDEO_ENCODER_FORMATS"):
+        return [checks.Error("No DJANGO_VIDEO_ENCODER_FORMATS defined in settings")]
+    errors = []
+    encoder_list = [format for format, __ in Format.VIDEO_CODEC_CHOICES]
+    formats_list = settings.DJANGO_VIDEO_ENCODER_FORMATS
+    for format in formats_list:
+        if "video_codec" not in format:
+            errors.append(
+                checks.Error(f"Format dict {format} has no defined `video_codec`")
+            )
+            continue
+        if not format["video_codec"] in encoder_list:
+            errors.append(
+                checks.Error(
+                    f"Requested codec {format['video_codec']} is not defined in "
+                    f"{encoder_list}"
+                )
+            )
+    duplicates = [
+        format for n, format in enumerate(formats_list) if format in formats_list[:n]
+    ]
+    for duplicate in duplicates:
+        errors.append(
+            checks.Error(
+                f"Format dict {duplicate} is duplicated in "
+                f"settings.DJANGO_VIDEO_ENCODER_FORMATS"
+            )
+        )
```

Comment:
@anrie and also @hackbert404 with pleasure:

Looks cool right? Unfortunately it doesn't work...

If you install the branch with an egg, keep old/wrong settings, the `manage.py check` won't do a thing. Add a breaking point, it will never go there... Any ideas?

Reply:
OK, I finally got it! It was really not clear from the django docs how to do, but here is: https://github.com/jonasundderwolf/django-video-encoder/pull/6/commits/b3e82e1fc7270d32ddca9349f2ee4ca36336bb47